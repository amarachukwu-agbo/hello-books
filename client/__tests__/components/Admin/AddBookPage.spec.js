import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import request from 'superagent/lib/client';
import nocker from 'superagent-nock';
import ConnectedAddBookPage, { AddBookPage } from '../../../components/Admin/AddBookPage';


const props = {
  addBook: sinon.spy(() => Promise.resolve()),
};
const files = [
      {
        lastModified: 1528116253099,
        lastModifiedDate: 'Mon Jun 04 2018 13:44:13 GMT+0100 (West Africa Standard Time)',
        name: 'adolfo-felix-586654-unsplash.jpg',
        preview:'blob:http://127.0.0.1:8080/33e8015f-6038-49ff-a8fd-8fedd140bf1a',
        size:5610114,
        type:'image/jpeg',
        webkitRelativePath: '',
      }
    ];
const mockStore = configureMockStore([thunk]);
const store = mockStore({});
const nock = nocker(request);

nock('https://api.cloudinary.com/v1_1/test-url')
   .post('/image/upload')
   .reply(200, {
    	secure_url: 'http://res.cloudinary.com/ama-hello-books-v2/image/upload/v1531091938/accessory-bracelets-earrings-949590_zawwj1.jpg',
   });


describe('<AddBookPage />', () => {
  it('should render correctly', () => {
    const tree = shallow(<AddBookPage {...props}/>);
    expect(tree).toMatchSnapshot();
  });
  it('calls submitForm()', () => {
    const values = {
      author: 'Author',
      title: 'A book',
      quantity: 5,
      description: 'A new book',
      subject: {
        value: 'Romance',
      }
    };
    const wrapper = shallow(<AddBookPage {...props}/>);
    wrapper.setState({
      cloudinaryUrl: 'http://res.cloudinary.com/test-url/image/upload/v1531091938/accessory-bracelets-earrings-949590_zawwj1.jpg',
    });
    sinon.spy(wrapper.instance(), 'submitForm');
    wrapper.instance().submitForm(values);
    expect(wrapper.instance().submitForm.calledOnce)
      .toEqual(true);
    expect(props.addBook.called).toEqual(true);
  });
  it('calls handleDrop', ()=> {
    const wrapper = shallow(<AddBookPage {...props}/>);
    sinon.spy(wrapper.instance(), 'handleDrop');
    wrapper.instance().handleDrop(files);
    expect(wrapper.instance().handleDrop.called).toEqual(true);
    expect(wrapper.state().uploadedFileCloudinaryUrl)
      .toEqual('http://res.cloudinary.com/ama-hello-books-v2/image/upload/v1531091938/accessory-bracelets-earrings-949590_zawwj1.jpg');
    expect(wrapper.state().uploadError)
      .toEqual('');
    expect(wrapper.state().isUploadingImage)
      .toEqual(false);
  });
});

describe('<ConnectedAddBookPage />', () => {
  it('should render successfully', () => {
    const connectedWrapper = 
      shallow(<ConnectedAddBookPage store = { store }/>);
      expect(connectedWrapper.length).toBe(1);
  });
});

