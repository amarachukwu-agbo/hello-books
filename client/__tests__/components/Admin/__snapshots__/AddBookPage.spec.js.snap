// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddBookPage /> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddBookPage
    addBook={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col s2 m3 l4"
        />,
        <div
          className="card-panel add-book col s8 m6 l4"
        >
          <ReduxForm
            addBook={[Function]}
            handleDrop={[Function]}
            isUploadingImage={false}
            submitForm={[Function]}
            uploadError={undefined}
            uploadedFile={undefined}
            uploadedFileCloudinaryUrl=""
          />
        </div>,
        <div
          className="col s2 m3 l4"
        />,
      ],
      "className": "row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "col s2 m3 l4",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ReduxForm
            addBook={[Function]}
            handleDrop={[Function]}
            isUploadingImage={false}
            submitForm={[Function]}
            uploadError={undefined}
            uploadedFile={undefined}
            uploadedFileCloudinaryUrl=""
          />,
          "className": "card-panel add-book col s8 m6 l4",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "addBook": [Function],
            "handleDrop": [Function],
            "isUploadingImage": false,
            "submitForm": [Function],
            "uploadError": undefined,
            "uploadedFile": undefined,
            "uploadedFileCloudinaryUrl": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "col s2 m3 l4",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col s2 m3 l4"
          />,
          <div
            className="card-panel add-book col s8 m6 l4"
          >
            <ReduxForm
              addBook={[Function]}
              handleDrop={[Function]}
              isUploadingImage={false}
              submitForm={[Function]}
              uploadError={undefined}
              uploadedFile={undefined}
              uploadedFileCloudinaryUrl=""
            />
          </div>,
          <div
            className="col s2 m3 l4"
          />,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "col s2 m3 l4",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ReduxForm
              addBook={[Function]}
              handleDrop={[Function]}
              isUploadingImage={false}
              submitForm={[Function]}
              uploadError={undefined}
              uploadedFile={undefined}
              uploadedFileCloudinaryUrl=""
            />,
            "className": "card-panel add-book col s8 m6 l4",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "addBook": [Function],
              "handleDrop": [Function],
              "isUploadingImage": false,
              "submitForm": [Function],
              "uploadError": undefined,
              "uploadedFile": undefined,
              "uploadedFileCloudinaryUrl": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "col s2 m3 l4",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
